File: WebApi/Models/VehicleModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace WebApi.Models {

	public class VehicleModel {

		public int VehicleId { get; set; }
		[Required]
		[MaxLength(16)]
		public string VehicleNumber { get; set; }
		[Required]
		public DateTime ManufactureDate { get; set; }
		[Required]
		public int Mileage { get; set; }
		[Required]
		public bool IsProper { get; set; }
		[Required]
		public bool IsAvailable { get; set; }
		public int? AtBranchId { get; set; }
		[Required]
		public int CarClassId { get; set; }
		[Required]
		public int GearTypeId { get; set; }
		[Required]
		public int ModelId { get; set; }
	}
}

File: AngularClient/src/app/ts/classes/models/vehicle.model.ts
import { VehicleI, BranchI, CarClassI, GearTypeI, VehicleModelI } from '../../interfaces/model.interfaces';

export class Vehicle implements VehicleI {
	public vehicleId: number
	public vehicleNumber: string
	public manufactureDate: Date
	public mileage: number
	public isProper: boolean
	public isAvailable: boolean
	public atBranchId: number
	public carClassId: number
	public gearTypeId: number
	public modelId: number
	public picturePath: string
	public isActive: boolean
	constructor(
		public atBranch: BranchI,
		public carClass: CarClassI,
		public gearType: GearTypeI,
		public model: VehicleModelI,
	) { }
}
